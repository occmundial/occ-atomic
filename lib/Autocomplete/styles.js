import idx from 'idx';
import { createUseStyles } from 'react-jss';

import spacing from '../subatomic/spacing';

export default createUseStyles({
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: props => idx(props, _ => _.textfieldProps.label) || idx(props, _ => _.textfieldProps.lockHeight) ? spacing.xLarge : 40,
        left: 0,
        width: '100%'
    }
});
