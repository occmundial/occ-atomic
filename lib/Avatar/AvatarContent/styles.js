import fonts from '../../tokens/fonts.json';
import colors from '../../tokens/colors.json';
import borderRadius from '../../tokens/borderRadius.json';
import legacyFont from '../../subatomic/fonts';

const { avatar, text, border } = colors;
const transition = 'all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s';

const font = fonts['body-regular-strong'];

export default {
	wrap: {
		position: 'relative',
		width: '100%',
		height: '100%'
	},
	cont: {
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		borderRadius: borderRadius['br-full'],
		overflow: 'hidden',
		background: avatar.bg.default,
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center'
	},
	capital: {
		display: 'inline-block',
		font: `${font.fontWeight} ${font.fontSize}/${font.lineHeight} ${legacyFont.body}`,
		color: text.indigo.primary,
		gridRowStart: '1',
		gridColumnStart: '1',
		textAlign: 'center'
	},
	disabled: {
		pointerEvents: 'none',
		background: avatar.bg.disabled,
		'& $camera': {
			opacity: 1
		}
	},
	disabledCapital: {
		color: text.corp.disabled
	  },
	clickable: {
		transition,
			'&:hover': {
				background: avatar.bg.hover
			},
			'&:active': {
				background: avatar.bg.active
			}
	  },
	editable: {
		'&:hover $camera': {
			opacity: 1
		},
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	},
	person: {
		gridRowStart: 1,
		gridColumnStart: 1,
		transition
	},
	  photo: {
		border: `1px solid ${border.default.subtle}`
	},
	hoverPhoto: {
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	  },
	camera: {
		gridRowStart: 1,
		gridColumnStart: 1,
		opacity: 0,
		transition
	},
	overlay: {
		borderRadius: '50%',
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center',
		transition,
		border: `1px solid transparent`
	  }
};
