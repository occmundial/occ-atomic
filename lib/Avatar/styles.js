import { createUseStyles } from 'react-jss';

export default createUseStyles({
    circle: {
        position:'relative',
        width: props => props.size ? props.size : 70,
        height: props => props.size ? props.size : 70,
        borderRadius:'50%',
        overflow:'hidden'
    }
});
