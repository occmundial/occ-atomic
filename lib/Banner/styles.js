import spacing from '../subatomic/spacing';
import colors from '../subatomic/colors';
import grid from '../subatomic/grid';

export default {
  container: { padding: spacing.small, backgroundColor: colors.ink },
  textBanner: { display: 'inline-block' },
  textFlex: {
    textAlign: 'center',
    [`@media screen and (max-width:${grid.xs}px)`]: { textAlign: 'start' },
  },
  iconFlex: { [`@media screen and (max-width:${grid.xs}px)`]: { padding: spacing.small } },
  icon: { cursor: 'pointer' },
};
