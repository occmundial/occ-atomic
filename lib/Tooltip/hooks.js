import { useEffect, useCallback, useMemo, useState, useRef } from 'react';

export function useOpenTooltipState(
	openExternal,
	setOpenExternal,
	closeDelay
  ) {
	const { throttle, cancel } = useThrottle({ trailing: true, leading: false })
  
	const [openInternal, setOpenInternal] = useState(false)
  
	const open = useMemo(
	  () => openExternal ?? openInternal,
	  [openExternal, openInternal]
	)
  
	const setOpenSource = useMemo(
	  () =>
		openExternal !== undefined
		  ? setOpenExternal ?? (() => {})
		  : setOpenInternal,
	  [openExternal, setOpenExternal, setOpenInternal]
	)
  
	const setOpenSourceDelay = useCallback(
	  (show) => {
		throttle({
		  throttleTime: closeDelay,
		  callback: () => setOpenSource(show)
		})
	  },
	  [closeDelay, setOpenSource, throttle]
	)
  
	function setOpen(openValue) {
	  if (openValue) {
		if (open) cancel()
		setOpenSource(true)
	  } else {
		closeDelay > 0 ? setOpenSourceDelay(false) : setOpenSource(false)
	  }
	}
  
	useEffect(() => {
	  if (closeDelay <= 0) return
	  if (openExternal === false) {
		cancel()
	  } else {
		setOpenSourceDelay(false)
	  }
	}, [openExternal, setOpenSourceDelay, setOpenSource, closeDelay, cancel])
  
	return [open, setOpen]
  }

export function useThrottle(defaults = {}) {
	const timeout = useRef(null)
	const lastCallback = useRef(null)
	const date = useRef(null)
  
	const cancel = useCallback(() => {
	  if (!timeout.current) return
	  window.clearTimeout(timeout.current)
	  timeout.current = null
	  date.current = null
	}, [])
  
	const flush = useCallback(() => {
	  if (!lastCallback.current) return
	  lastCallback.current()
	  lastCallback.current = null
	  date.current = null
	  cancel()
	}, [cancel])
  
	const throttle = useCallback(
	  ({ callback, throttleTime, leading, trailing }) => {
		if (timeout.current) {
		  return (lastCallback.current = callback ?? defaults.callback ?? null)
		} else {
		  date.current = new Date()
		  if (defaults?.leading ?? leading ?? true) {
			callback()
		  } else {
			lastCallback.current = callback
		  }
		}
		timeout.current = window.setTimeout(() => {
		  defaults?.trailing ?? trailing ? flush() : cancel()
		}, throttleTime ?? defaults.throttleTime ?? 1000)
	  },
	  [
		cancel,
		defaults.callback,
		defaults?.leading,
		defaults.throttleTime,
		defaults?.trailing,
		flush
	  ]
	)
  
	useEffect(() => cancel, [cancel])
  
	return {
	  throttle,
	  flush,
	  cancel,
	  isActive: () => !!date.current,
	  date
	}
  }
