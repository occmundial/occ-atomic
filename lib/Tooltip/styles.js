import colors from '../subatomic/colors';
import newColors from '../tokens/colors.json';
import fonts from '../tokens/fonts.json';
import spacing from '../tokens/spacing.json';
import borderRadius from '../tokens/borderRadius.json';

import { parseFontValue } from '../Text/styles';

const { infoLight, infoText, info, white } = colors;

export default {
  activator: {
    display: 'inline-block'
  },
  tooltip: {
    padding: [spacing['size-2'], spacing['size-3']],
    borderRadius: borderRadius['br-xs']
  },
  content: {
    display: 'flex',
    gap: spacing['size-1'],
    alignItems: 'center',
    justifyContent: 'center'
  },
  info: {
    background: infoLight,
    color: infoText
  },
  dark: {
    background: newColors.bg.neutral,
    color: white,
    boxShadow: `inset 0 0 0 1px ${newColors.border.inverse.subtle}`
  },
  light: {
    background: newColors.bg.surface.default,
    color: info,
    '&:after': {
      boxShadow: `inset -2px -2px 0 -1px ${newColors.border.default.subtle}`
    },
    boxShadow: `inset 0 0 0 1px ${newColors.border.default.subtle}`
  },
  purple: {
    background: info,
    color: white
  },
  text: {
    font: parseFontValue(fonts['body-x-small']),
    margin: 0
  }
};
