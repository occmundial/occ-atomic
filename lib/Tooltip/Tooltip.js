import React, { useRef, useMemo, Fragment } from 'react';
import PropTypes from 'prop-types';
import {
	useFloating,
	autoUpdate,
	offset,
	useHover,
	useRole,
	useInteractions,
	FloatingPortal,
	arrow,
	FloatingArrow,
	size
} from '@floating-ui/react';
import colors from '../subatomic/colors';
import { useOpenTooltipState } from './hooks';

const { infoLight, white, grey900, info } = colors;

const themes = {
	DARK: 'dark',
	LIGHT: 'light',
	INFO: 'info',
	PURPLE: 'purple'
};

const colorsArrow = {
	[themes.DARK]: grey900,
	[themes.LIGHT]: white,
	[themes.INFO]: infoLight,
	[themes.PURPLE]: info
};

/** Tooltip component to display messages within a container. */
function Tooltip({
	classes,
	open: openProp,
	children,
	text,
	theme,
	openOnHover = false,
	closeDelay = 4000,
	zIndex = 10,
	placement = 'top',
	showArrow = true,
	className = {},
	fit = false,
	width = 220,
	strategy = 'absolute',
	onChange
}) {
	const arrowRef = useRef(null);

	const [open, setOpen] = useOpenTooltipState(openProp, onChange, closeDelay);

	const getMiddlewares = useMemo(() => {
		const middlewares = [offset(16)];
		showArrow && middlewares.push(arrow({ element: arrowRef, padding: 16 }));
		const sizeMiddleware = size({
			apply({ elements, rects, availableWidth }) {
				const styles = {};
				if (fit) {
					styles.width = `${rects.reference.width}px`;
				} else {
					styles.maxWidth = `${availableWidth}px`;
					if (width)
						styles.width = typeof width === 'string' ? width : `${width}px`;
					else styles.width = '';
				}
				Object.assign(elements.floating.style, styles);
			}
		});
		sizeMiddleware.name = `size-${fit}-${width}`;
		middlewares.push(sizeMiddleware);
		return middlewares;
	}, [showArrow, fit, width]);

	const { refs, floatingStyles, context } = useFloating({
		open: open,
		onOpenChange: setOpen,
		placement,
		strategy,
		whileElementsMounted: autoUpdate,
		middleware: getMiddlewares
	});

	const hover = useHover(context, {
		enabled: openOnHover
	});
	const role = useRole(context, { role: 'tooltip' });

	const { getReferenceProps, getFloatingProps } = useInteractions([
		hover,
		role
	]);

	return (
		<Fragment>
			<div
				ref={refs.setReference}
				{...getReferenceProps()}
				className={`${classes.activator} ${
					className.hasOwnProperty('activator') && className.activator
				}`}
			>
				{children}
			</div>

			<FloatingPortal>
				{open && (
					<div
						className={`${classes.tooltip} ${
							className.hasOwnProperty('tooltip') && className.tooltip
						} ${classes[theme] || classes.purple}`}
						ref={refs.setFloating}
						style={{ ...floatingStyles, zIndex, position: strategy }}
						{...getFloatingProps()}
					>
						{text}
						{showArrow && (
							<FloatingArrow
								ref={arrowRef}
								context={context}
								fill={colorsArrow[theme] || colorsArrow[themes.PURPLE]}
								width={14}
								height={10}
							/>
						)}
					</div>
				)}
			</FloatingPortal>
		</Fragment>
	);
}

Tooltip.propTypes = {
	classes: PropTypes.object,
	/** It accepts any type of the followings boolean,null,undefined,ReactFragment,ReactPortal,ReactChild. */
	children: PropTypes.node,
	/** This property can be used to set the theme of the component. The possible values for the theme property are info, light, dark, and purple. */
	theme: PropTypes.oneOf(['light', 'dark', 'purple', 'info']),
	/** This property can be used to control whether or not the component is displayed. */
	open: PropTypes.bool,
	/** This property can be used to set the text that is displayed by the component. */
	text: PropTypes.string.isRequired,
	/** Designates whether the tooltip should open on activator hover. */
	openOnHover: PropTypes.bool,
	/** Time on millisecond that it will take to close the tooltip. */
	closeDelay: PropTypes.number,
	/** Sets the Tooltip stack order. */
	zIndex: PropTypes.number,
	/** Allows to specify the placement of Tooltip. Available options are top, top-start, top-end, bottom, bottom-start, bottom-end, left, right. */
	placement: PropTypes.string,
	/** Allows to specify classes to activador and tooltip. Object has the next properties: activator(string), tooltip(string). */
	className: PropTypes.object,
	/** This property can be used to control whether or not an arrow is displayed. */
	showArrow: PropTypes.bool,
	/** This property allows the Tooltip to be adjusted to the length of the associated component. */
	fit: PropTypes.bool,
	/** The "Width" property of the Tooltip enables modification of its width and accepts a number (e.g., 220), a string (e.g., '220px' or 'auto') or null as its value. */
	width: PropTypes.number,
	/** The CSS position property to use to compute the tooltip position */
	strategy: PropTypes.oneOf(['absolute', 'fixed']),
	/** Callback fired when the Tooltip show state changes. */
	onChange: PropTypes.func
};

export default Tooltip;
