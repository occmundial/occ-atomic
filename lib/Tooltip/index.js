import injectSheet from 'react-jss';
import Tooltip from './Tooltip';
import styles from './styles';

export default injectSheet(styles)(Tooltip);
