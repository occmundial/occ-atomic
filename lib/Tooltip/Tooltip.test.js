import React from 'react';
import { shallow } from 'enzyme';
import Tooltip from './Tooltip';
import styles from './styles';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe('Tooltip', () => {
	it('matches the snapshot', () => {
		const wrapper = shallow(
			<Tooltip text="test" open theme="dark" classes={classes} />
		);
		expect(wrapper.html()).toMatchSnapshot();
	});
});

describe('Tooltip styles', () => {
	it('matches the snapshot', () => {
		expect(styles).toMatchSnapshot();
	});
});
