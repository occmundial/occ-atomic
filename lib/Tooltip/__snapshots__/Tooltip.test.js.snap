// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip matches the snapshot 1`] = `"<div class=\\"activator \\"></div>"`;

exports[`Tooltip styles matches the snapshot 1`] = `
Object {
  "activator": Object {
    "display": "inline-block",
  },
  "content": Object {
    "alignItems": "center",
    "display": "flex",
    "gap": "4px",
    "justifyContent": "center",
  },
  "dark": Object {
    "background": "#000",
    "boxShadow": "inset 0 0 0 1px rgba(255,255,255,0.2)",
    "color": "#ffffff",
  },
  "info": Object {
    "background": "#efebf7",
    "color": "#5736ab",
  },
  "light": Object {
    "&:after": Object {
      "boxShadow": "inset -2px -2px 0 -1px #EDEDF1",
    },
    "background": "#fff",
    "boxShadow": "inset 0 0 0 1px #EDEDF1",
    "color": "#5736ab",
  },
  "purple": Object {
    "background": "#5736ab",
    "color": "#ffffff",
  },
  "text": Object {
    "font": "300 12px/1.5 'OccText', sans-serif",
    "margin": 0,
  },
  "tooltip": Object {
    "borderRadius": "4px",
    "padding": Array [
      "8px",
      "12px",
    ],
  },
}
`;
